ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.cpp

```diff
@@ -626,8 +644,16 @@ std::tuple<ttnn::Tensor, ParallelConfig, bool> shard_or_reshard_tensor_if_requir
             }
             input_tensor = resharded_input_tensor;
         } else {
-            input_tensor = ttnn::operations::core::to_device(
-                input_tensor, device, input_tensor_sharded_memory_config);
+            if (mm_conv && input_tensor.layout() == Layout::ROW_MAJOR &&
+                parallel_config.shard_scheme != TensorMemoryLayout::HEIGHT_SHARDED) {
```

Comment:
What about width sharded?

Given the comment I'm wondering if the two checks should be ... == .. BLOCK_SHARDED to make it less confusing.

Reply:
This is more generic in this case we won't get width shared as mm doesn't support it and auto shard Is not going to set it

Reply:
Ok.